### PR TITLE
Schema update: computed

### LINKED ISSUE #65: implement `computed` (read-only) attributes within Schema
https://github.com/netascode/terraform-provider-fmc/pull/40#discussion_r1659363907

Extend the schema to support `Computed` (read-only) attributes. We have implemented this here already: https://github.com/CiscoDevNet/terraform-provider-ise/blob/479742e9945413edf01efb935b2f63582315b9e6/gen/schema/schema.yaml#L43 .

Mark the `type` attributes, and similar, as computed in the resource definitions.

### PR DESCRIPTION
**Pull request by rchrabas**
Adds support for new schema field - `computed` as attribute to consider attribute read_only/computed.
Field value is assigned from API response.
For testing purposes, `Host`, `Hosts` and `SGT` objects were modified, which covers single and bulk resources.

This resolves #65

### CODE REVIEW

File: gen/schema/schema.yaml [hunk 1]
```diff
@@ -48,6 +48,7 @@ attribute:
   string_patterns: list(str(), required=False) # List of regular expressions that the string must match, only relevant if type is "String"
   string_min_length: int(required=False) # Minimum length of a string, only relevant if type is "String"
   string_max_length: int(required=False) # Maximum length of a string, only relevant if type is "String"
+  read_only: any(str(), int(), bool(), required=False) # Sets field as read-only (computed) with the provided value. This value may be overriden by value returned from API
```
Comment (danischm) [inline gen/schema/schema.yaml:51]: Could we rename this to `computed` and make it boolean to align with what we already have for ISE? I don't think we need the expected value here as it should always be retrieved from the API response.

File: gen/definitions/host.yaml [hunk 2]
```diff
@@ -12,10 +12,8 @@ attributes:
   - model_name: type
     type: String
     description: Type of the object; this value is always 'Host'.
-    default_value: Host
-    requires_replace: true
+    read_only: Host
     exclude_example: true
```
Comment (danischm) [inline gen/definitions/host.yaml:16]: `exclude_example` should not be required for read-only attributes. If necessary, we should update the templates rendering the examples to always exclude those attributes.

File: gen/schema/schema.yaml (hunk 1 above)
Comment (rchrabas) [reply gen/schema/schema.yaml]: changed

File: gen/definitions/host.yaml (hunk 2 above)
Comment (rchrabas) [reply gen/definitions/host.yaml]: changed

File: gen/templates/model.go [hunk 3]
```diff
@@ -267,6 +269,8 @@ func (data *{{camelCase .Name}}) fromBody(ctx context.Context, res gjson.Result)
 	} else {
 		{{- if .DefaultValue}}
 		data.{{toGoName .TfName}} = types.{{.Type}}Value({{if eq .Type "String"}}"{{end}}{{.DefaultValue}}{{if eq .Type "String"}}"{{end}})
+		{{- else if .ReadOnly}}
```
[the remark this answers is not in the document]
Comment (rchrabas) [reply gen/templates/model.go]: changed

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/templates/model.go [hunk 4]
```diff
@@ -340,7 +342,7 @@ func (data *{{camelCase .Name}}) fromBodyPartial(ctx context.Context, res gjson.
 	{{- range .Attributes}}
 	{{- if and (not .Value) (not .WriteOnly) (not .Reference)}}
 	{{- if or (eq .Type "String") (eq .Type "Int64") (eq .Type "Float64") (eq .Type "Bool")}}
-	if value := res.Get("{{range .DataPath}}{{.}}.{{end}}{{.ModelName}}"); value.Exists(){{if not .ResourceId}} && !data.{{toGoName .TfName}}.IsNull(){{end}} {
+	if value := res.Get("{{range .DataPath}}{{.}}.{{end}}{{.ModelName}}"); value.Exists(){{if and (not .ResourceId)}} && !data.{{toGoName .TfName}}.IsNull(){{end}} {
```
[the remark this answers is not in the document]
Comment (rchrabas) [reply gen/templates/model.go]: removed

File: gen/templates/data_source_test.go [hunk 5]
```diff
@@ -284,6 +289,7 @@ func testAccDataSourceFmc{{camelCase .Name}}Config() string {
 func testAccNamedDataSourceFmc{{camelCase .Name}}Config() string {
 	config := `resource "fmc_{{snakeCase $name}}" "test" {` + "\n"
 	{{- range  .Attributes}}
+	{{- if .Computed}}{{- continue }}{{- end}}
```
[the remark this answers is not in the document]
Comment (rchrabas) [reply gen/templates/data_source_test.go:294]: Implemented

**Review by danischm — APPROVED** [review]

**Pull request merged**